packages/core/src/amazonqFeatureDev/errors.ts

```diff
@@ -57,6 +57,44 @@ export class SelectedFolderNotInWorkspaceFolderError extends ToolkitError {
     }
 }
 
+export class GuardrailsException extends ToolkitError {
+    constructor() {
+        super("I'm sorry, I'm having trouble generating your code. Please try again. ", {
+            code: 'GuardrailsException',
+        })
+    }
+}
+
+export class PromptRefusalException extends ToolkitError {
+    constructor() {
+        super(
+            'I\'m sorry, I can\'t generate code for your request. Please make sure your message and code files comply with the <a href="https://aws.amazon.com/machine-learning/responsible-ai/policy/">AWS Responsible AI Policy.</a>',
+            {
+                code: 'PromptRefusalException',
+            }
+        )
+    }
+}
+
+export class EmptyPatchException extends ToolkitError {
+    constructor() {
+        super("I'm sorry, I'm having trouble generating your code. Please try again.", {
+            code: 'EmptyPatchException',
+        })
+    }
+}
+
+export class ThrottlingException extends ToolkitError {
+    constructor() {
+        super(
+            "I'm sorry, I'm experiencing high demand at the moment and can't generate your code. This attempt won't count toward usage limits. Please try again.",
```

Comment:
Do these new errors map to service-side errors? If so, can we define a `FeatureDevServiceError extends ToolkitError` instead of defining a new error class for every single failure mode?

Reply:
The reason that we have the mapping is to use and respond differently for those if we need. Some require retries, other don't, so have them separately help map those internally. But I get your point. One thing that we can make is a FeatureDevServiceError and send over the message we want when we're creating. We use this in all that have the same behaviour expected (message with retries).